system/core/pages.php

```diff
@@ -601,6 +601,19 @@ function wikify_link($node) {
 			$node->text('');
 	}
 
+	function figcaption($element) {
+		foreach ($element->findXPath(".//img[@longdesc]") as $img) {
+			figcaption_img($img);
+		}
+	}
+
+	function figcaption_img($img) {
+		$text = $img->getAttribute('longdesc');
+		/** @var DOMWrap\Element $html */
+        $caption = new DOMWrap\Element('figcaption', $text);
+		$img->parentNode->insertBefore($caption, $img->nextSibling);
```

Comment:
Doesn't this need a `<figure>` tag as well to group the caption with the image? https://developer.mozilla.org/en-US/docs/Web/HTML/Element/figcaption

Reply:
Ah, I didn't know! fixed